src/Bedrock/Entity/BedrockWorkflow.workflow.cs

```diff
@@ -0,0 +1,164 @@
+                // AWS Bedrock.
+
+                // We summarize the chat to date and use that as input to the
+                // new workflow
+                if (ConversationHistory.Count >= continueAsNewPerTurns)
+                {
+                    // Summarize the conversation to date using Amazon Bedrock
+                    var summaryResult = await Workflow.ExecuteActivityAsync(
+                        (IBedrockActivities activities) => activities.PromptBedrockAsync(new(PromptSummaryFromHistory())),
+                        new()
+                        {
+                            StartToCloseTimeout = TimeSpan.FromSeconds(20),
+                        });
+
+                    ConversationSummary = summaryResult.Response;
+                    Workflow.Logger.LogInformation("Continuing as new due to {ContinueAsNewPerTurns} conversational turns.", continueAsNewPerTurns);
+
+                    throw Workflow.CreateContinueAsNewException<BedrockWorkflow>(workflow =>
+                        workflow.RunAsync(new(ConversationSummary, promptQueue)));
+                }
+            }
+
+            // If end chat signal was sent
+            if (chatEnded)
+            {
+                // The workflow might be continued as new without any
+                // chat to summarize, so only call Bedrock if there
+                // is more than the previous summary in the history.
+                if (ConversationHistory.Count > 1)
+                {
+                    var summaryResult = await Workflow.ExecuteActivityAsync(
+                        (IBedrockActivities activities) => activities.PromptBedrockAsync(new(PromptSummaryFromHistory())),
+                        new()
+                        {
+                            StartToCloseTimeout = TimeSpan.FromSeconds(20),
+                        });
+
+                    ConversationSummary = summaryResult.Response;
+                    Workflow.Logger.LogInformation("Chat ended. Conversation summary:\n{ConversationSummary}", ConversationSummary);
+                    return new BedrockWorkflowResult(ConversationHistory);
+                }
+            }
+        }
+    }
+
+    [WorkflowQuery]
+    public string? ConversationSummary { get; private set; }
+
+    [WorkflowQuery]
+    public Collection<ConversationEntry> ConversationHistory { get; } = new();
+
+    [WorkflowSignal]
+    public Task UserPromptAsync(BedrockUserPromptSignal signal)
+    {
+        // Chat timed out but the workflow is waiting for a chat summary to be generated
+        if (chatEnded)
+        {
+            Workflow.Logger.LogWarning("Message dropped due to chat closed: {Prompt}", signal.Prompt);
+            return Task.CompletedTask;
+        }
+
+        promptQueue.Enqueue(signal.Prompt);
+        return Task.CompletedTask;
+    }
+
+    [WorkflowSignal]
+    public Task EndChatAsync()
+    {
+        chatEnded = true;
+        return Task.CompletedTask;
+    }
```

Comment:
```suggestion
    public async Task EndChatAsync() => chatEnded = true;
```

If you want to save the lines, but not required (and I know making a method `async` that doesn't do `await`ing is usually frowned upon in C# but not for workflows since we require signals be async)